providers/org/plos/harvester.py

```diff
@@ -0,0 +1,46 @@
+from furl import furl
+from lxml import etree
+
+from project.settings import PLOS_API_KEY
+
+from share import Harvester
+
+
+class PLOSHarvester(Harvester):
+
+    url = 'http://api.plos.org/search'
+    MAX_ROWS_PER_REQUEST = 999
+
+    def do_harvest(self, start_date, end_date):
+        start_date = start_date.date()
+        end_date = end_date.date()
+
+        return self.fetch_rows(furl(self.url).set(query_params={
+            'q': 'publication_date:[{}T00:00:00Z TO {}T23:59:59Z]'.format(start_date.isoformat(), end_date.isoformat()),
```

Comment:
Yep

On Wed, Jun 29, 2016 at 4:41 PM, Casey Rollins <notifications@github.com>
wrote:

> In providers/org/plos/harvester.py
> <https://github.com/CenterForOpenScience/SHARE/pull/187#discussion_r69023343>
> :
>
> > +from project.settings import PLOS_API_KEY
> > +
> > +from share import Harvester
> > +
> > +
> > +class PLOSHarvester(Harvester):
> > +
> > +    url = 'http://api.plos.org/search'
> > +    MAX_ROWS_PER_REQUEST = 999
> > +
> > +    def do_harvest(self, start_date, end_date):
> > +        start_date = start_date.date()
> > +        end_date = end_date.date()
> > +
> > +        return self.fetch_rows(furl(self.url).set(query_params={
> > +            'q': 'publication_date:[{}T00:00:00Z TO {}T23:59:59Z]'.format(start_date.isoformat(), end_date.isoformat()),
>
> @chrisseto <https://github.com/chrisseto>, do you mean something along
> these lines start_date.format('YYYY-MM-DDT00:00:00') + 'Z'?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CenterForOpenScience/SHARE/pull/187/files/946743581ed2c0bc8f5ec9aa113fbb80cf33cf52#r69023343>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AFRs6TLrQQ2_ufpL1pk9mBUs2_tOtbUeks5qQth0gaJpZM4JBiS2>
> .
>